bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/admin/ContentAdminResource.java

```diff
@@ -0,0 +1,155 @@
+@Path( "/api/admin/content" )
+@ApplicationScoped
+@REST
+public class ContentAdminResource
+                implements IndyResources
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private TrackingContentController controller;
+
+    @Inject
+    private ResponseHelper responseHelper;
+
+    // TODO need to disable folo {id}/repo/zip endpoint
+    @ApiOperation(
+                    "Retrieve the content referenced in a tracking record as a ZIP-compressed Maven repository directory." )
+    @ApiResponses( { @ApiResponse( code = 200, response = File.class, message = "ZIP repository content" ),
+                    @ApiResponse( code = 404, message = "No such tracking record" ) } )
+    @Path( "/repo/zip" )
+    @POST
+    @Produces( application_zip )
+    public File getZipRepository( @Context final UriInfo uriInfo, final TrackedContent record )
+    {
+        try
+        {
+            return controller.renderRepositoryZip( record );
+        }
+        catch ( IndyWorkflowException e )
+        {
+            responseHelper.throwError( e );
+        }
+
+        return null;
+    }
+
+    // TODO need to disable folo recalculate endpoint
+    @ApiOperation( "Recalculate sizes and checksums for every file listed in a tracking record." )
+    @ApiResponses( {
+                    @ApiResponse( code = 200, response = ContentTransferDTO.class, message = "Recalculated tracking report" ),
+                    @ApiResponse( code = 404, message = "No such tracking record can be found" ),
+                    @ApiResponse( code = 400, message = "All entries must belong to the same tracking Id" ) } )
+    @POST
+    @Path( "/tracking/recalculate" )
+    public Response recalculateEntrySet( @Context final UriInfo uriInfo, final Set<ContentTransferDTO> entries )
+    {
+        if ( entries == null )
+        {
+            return Response.status( Status.NOT_FOUND ).build();
+        }
+        String id = "";
+        String another_id = "";
+        for ( ContentTransferDTO entry : entries )
+        {
+            // Just get an id from an entry since all entries should belong to the same tracking id
+            id = entry.getTrackingKey().getId();
+            if ( !StringUtils.isNotBlank( another_id ) )
+            {
+                another_id = id;
+            }
+            else
+            {
+                if ( !another_id.equals( id ) )
+                {
+                    return Response.status( Status.BAD_REQUEST )
+                                   .entity( "All entries must belong to the same tracking Id" )
+                                   .build();
+                }
+                else
+                {
+                    break;
+                }
+            }
+        }
+        Response response;
+        try
+        {
+            final Set<TrackedContentEntry> newEntries =
```

Comment:
Is it possible to use ContentTrasferDTO to replace this? So we can remove the TrackedXXX out of this ContentAdminREsource.

Reply:
No it is not because we need this entire class to replace records in cassandra